exonum-java-binding-core/rust/src/utils/jni_cache.rs

```diff
@@ -0,0 +1,158 @@
+use jni::{
+    objects::{GlobalRef, JMethodID},
+    sys::{jint, JNI_VERSION_1_8},
+    JNIEnv, JavaVM,
+};
+use std::os::raw::c_void;
+
+static mut OBJECT_CACHED_GET_CLASS: Option<JMethodID> = None;
+static mut CLASS_CACHED_GET_NAME: Option<JMethodID> = None;
+static mut THROWABLE_CACHED_GET_MESSAGE: Option<JMethodID> = None;
+
+static mut UTA_CACHED_EXECUTE: Option<JMethodID> = None;
+static mut UTA_CACHED_INFO: Option<JMethodID> = None;
+static mut UTA_CACHED_VERIFY: Option<JMethodID> = None;
+
+static mut USA_CACHED_STATE_HASHES: Option<JMethodID> = None;
+static mut USA_CACHED_CONVERT_TRANSACTION: Option<JMethodID> = None;
+
+static mut CLASS_CACHED_ERROR: Option<GlobalRef> = None;
+static mut CLASS_CACHED_TEE: Option<GlobalRef> = None;
+
+#[allow(non_snake_case)]
+#[no_mangle]
```

Comment:
1. Probably this function could be moved to separate file since it's not semantically linked to JNI cache.
2. Not sure whether we need to implement JNI_On_Unload() to do any cleanup of cached data - everything should be destroyed with JVN shutdown.

Reply:
As we use this function only for cache - I can't see any reason to move it from here. We can always do it later.

Reply:
JNI_On_Unload() is unnecessary according to requirements